lib/utils.js

```diff
@@ -30,3 +32,51 @@ exports.basicAuth = function(username, password) {
 		next();
 	};
 };
+
+exports.findAvailablePort = function(app){
+
+  var port = (process.env.PORT || 3000);
+
+  console.log('');
+  // Check that default port is free, else offer to change
+  portScanner.findAPortNotInUse(port, port+50, '127.0.0.1', function(error, availablePort) {
+
+    if (port == availablePort){
+      app.listen(port);
+      console.log('Listening on port ' + port + '   url: http://localhost:' + port);
+    }
+    else {
+      // Default port in use - offer to change to available port
+      console.log("ERROR: Port " + port + " in use - you may have another prototype running.\n");
```

Comment:
Might be better using [`console.error`](https://nodejs.org/api/console.html#console_console_error_data) instead of `.log` here.